RainyRefresh/ONXRefreshControl.swift

```diff
@@ -0,0 +1,148 @@
+//
+//  BaseRefreshControl.swift
+//  BaseRefreshControl
+//
+
+import UIKit
+
+public class ONXRefreshControl: UIControl {
+    
+    private var forbidsOffsetChanges = false
+    private var forbidsInsetChanges = false
+    private var changingInset = false
+    private var refreshing = false
+    private var contentInset: UIEdgeInsets?
+    public var animationDuration = TimeInterval(0.33)
+    public var animationDamping = CGFloat(0.4)
+    public var animationVelocity = CGFloat(0.8)
+    //delay for finishing custom animation
+    public var delayBeforeEnd = 0.0
+    
+    private var expandedHeight: CGFloat {
+        return UIScreen.main.bounds.height / 6
+    }
+
+    // MARK: Initialization
+
+    public convenience init() {
+        self.init(frame: .zero)
+    }
+    
+    public required init?(coder aDecoder: NSCoder) {
+        super.init(coder: aDecoder)
+        commonInit()
+    }
+    
+    public override init(frame: CGRect) {
+        super.init(frame: frame)
+        commonInit()
+    }
+    
+    private func commonInit() {
```

Comment:
Let's get rid of `commonInit` and `setup` methods and leave this logic to subclasses. Reason - it is just adds boilerplate code to this class without any functionality.